### PR TITLE
LPD-18860 Functional automation for saving and editing of DSM cards sections

### PR DESCRIPTION
**Pull request by markocikos**
### References

- Parent story: [LPD-6623 Allow admins to save the definition for cards as visualization mode](https://issues.liferay.com/browse/LPD-6623)

### What is the goal of this PR?

In this PR, we are adding tests for saving and editing of DSM cards sections. 

Please see technical notes inline.

### What does it look like?

image

### CODE REVIEW

File: modules/test/playwright/tests/frontend-data-set-views-web/pages/view/visualization_modes/VisualizationModesPage.ts [hunk 1]
```diff
@@ -33,14 +58,56 @@ export class VisualizationModesPage {
...
+		sectionLabel,
+	}: {
+		container: Locator;
+		sectionLabel: string;
+	}) {
+		await container
+			.locator('tr')
+			.filter({has: this.page.getByText(sectionLabel)})
+			.getByTitle('Assign Field')
+			.click();
```
Comment (markocikos) [inline modules/test/playwright/tests/frontend-data-set-views-web/pages/view/visualization_modes/VisualizationModesPage.ts:72]: I merged `list` and `cards` tests, pages and fixures into `visualizationModes` tests, pages and fixures. There is a lot of overlap between them, a lot of code that can be reused. But, we now need `container` prop to isolate locator withing a tab. Visualization mode tabs are all rendered, with only one visible. That makes is a bit more awkward to work with locators, as we need to find unique elements within all three tabs. That's the main reason for explicit containers. Sidenote - modal is only ever one, that's why we can have one modal locator, and not three.

File: modules/test/playwright/tests/frontend-data-set-views-web/visualizationModes.spec.ts [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+		'LPS-164563': true,
+	}),
+	visualizationModesPageTest,
+	loginTest()
+);
+
+test.beforeEach(async ({dataSetManagerApiHelpers}) => {
+	await dataSetManagerApiHelpers.createDataSet({});
+	await dataSetManagerApiHelpers.createDataSetView({});
+});
```
Comment (markocikos) [inline modules/test/playwright/tests/frontend-data-set-views-web/visualizationModes.spec.ts:22]: `beforeAll` unfortunately does not work, as we need `page` context, used in `ApiHelpers`.

**Comment by markocikos** [comment]
ci:test:playwright

**Comment by markocikos** [comment]
ci:test:playwright

**Comment by markocikos** [comment]
ci:test:relevant

**Comment by markocikos** [comment]
ci:report:2544943411

**Comment by markocikos** [comment]
No unique failures in relevant, based on https://github.com/liferay-frontend/liferay-portal/pull/4025#issuecomment-1971442151

Just waiting for peer review.

**Review by juanjofgliferay — APPROVED** [review]
LGTM :rocket:

**Comment by markocikos** [comment]
ci:forward:force

**Comment by markocikos** [comment]
ci:forward:force

**Pull request closed without merging**